app/components/page-components/Chatbot/ChatbotBase.vue.ts

```diff
@@ -1,36 +1,46 @@
 import Vue from 'vue';
-import { Component, Prop } from 'vue-property-decorator';
+import { Component } from 'vue-property-decorator';
 import { ChatbotApiService, ChatbotCommonService } from 'services/chatbot/chatbot';
 import { Inject } from 'util/injector';
-import ToggleInput from 'components/shared/inputs/ToggleInput.vue';
 import Tabs from 'components/Tabs.vue';
-
+import DropdownMenu from 'components/shared/DropdownMenu.vue';
+import { inputComponents } from 'components/widgets/inputs';
 import {
   ChatbotPermissionsEnums,
-  ChatbotResponseTypes
+  ChatbotResponseTypes,
+  ChatbotPunishments,
 } from 'services/chatbot/chatbot-interfaces';
 
 @Component({
   components: {
-    ToggleInput,
+    ...inputComponents,
     Tabs,
+    DropdownMenu,
   }
 })
 export default class ChatbotBase extends Vue {
   @Inject() chatbotApiService: ChatbotApiService;
   @Inject() chatbotCommonService: ChatbotCommonService;
 
+  mounted() {
+    // pre-load them to switch between 2 windows
+    this.chatbotApiService.fetchDefaultCommands();
+    this.chatbotApiService.fetchLinkProtection();
+  }
+
   get chatbotPermissions() {
-    let permissions = Object.keys(ChatbotPermissionsEnums)
-      .reduce((a: any[], b: string) => {
+    let permissions = Object.keys(ChatbotPermissionsEnums).reduce(
+      (a: any[], b: string) => {
```

Comment:
You should define an interface in this file that describes the shape of that data you're returning here.  It would look something like:
```TypeScript
interface IChatbotPermissions {
  title: string;
  value: number;
}

(a: IChatbotPermissions[], b: string) => {
```

This is important because the return value of your getter is currently `any[]`, which makes it extremely difficult for other parts of the app to know how to use it, and you lose all type checking for anything that calls this.